### ISSUE TITLE
[RFC] How to use Contact on FSM Order

### ISSUE THREAD

**Issue by tafaRU** [OCA/field-service#1230]
Hello,

I'm writing to ask for clarification regarding the field Contact (`contact_id`) on FSM Order and how to use it.

Testing and diving into the code I checked these followings behaviours:

- it's only filled on FSM Order writing and the value used it's copied from Location Billed Customer (Ref. https://github.com/OCA/field-service/blob/3477731b5672040e8fe8a1c800bd18b757f86362/fieldservice_account_analytic/models/fsm_order.py#L39)
- if it's selected a Contact who has Primary Service Location, this latter has set as Location on FSM Order (Ref. https://github.com/OCA/field-service/blob/3477731b5672040e8fe8a1c800bd18b757f86362/fieldservice_account_analytic/models/fsm_order.py#L32)
- FSM Order can be billed to Bill Location or to Bill Contact corresponding respectively to Location Billed Customer (Ref. https://github.com/OCA/field-service/blob/fef357df9394c30b18bb3de913667cee6d7143e9/fieldservice_isp_account/models/fsm_order.py#L166) and to Contact field (Ref. https://github.com/OCA/field-service/blob/fef357df9394c30b18bb3de913667cee6d7143e9/fieldservice_isp_account/models/fsm_order.py#L156)

In the face of this what should be the best approach to cover the following scenario?

- create a FSM Order, choose a Location and Bill it to a Contact different to the one specified in Location Billed Customer

Thank you in advance.

**Comment by tafaRU** [comment]
The only solution that I found is the following:

1. create a Location
2. create a partner in which set the previous location as _Primary Service Location_
3. create a FSM ORDER and set _Contact_ to the previuos one
4. Bill To _Bill Contact_

But it seems to me a bit counterintuitive. So the question is: why _Contact_ field cannot be freely valued?

**Comment by max3903** [comment]
@tafaRU I will need to check further next week but the idea is you may have a lot of people involved in a location/order: the contact to call before and open the location, the owner, the one who will pay for the service, etc.. I will try to provide more details next week.

Thanks for your patience.

**Comment by tafaRU** [comment]
@max3903 you got a way to provide more details?
Thank you in advance.

**Comment by max3903** [comment]
@tafaRU It sounds like the issue you are having is with this onchange:

https://github.com/OCA/field-service/blob/16.0/fieldservice_account_analytic/models/fsm_order.py#L30

I understand you expect to select the location and any contact as the customer, correct?

What would be the intuitive steps?

- Create the location and select the customer to bill
- Create an order, select the location --> the customer gets set automatically based on the location

Would that work for you?

**Comment by tafaRU** [comment]
> [@tafaRU](https://github.com/tafaRU) It sounds like the issue you are having is with this onchange:
> 
> https://github.com/OCA/field-service/blob/16.0/fieldservice_account_analytic/models/fsm_order.py#L30

@max3903 yes, it is, I made https://github.com/OCA/field-service/pull/1322 indeed.
